wp-includes/js/service-worker-offline-commenting.js

```diff
@@ -0,0 +1,55 @@
+/* global ERROR_OFFLINE_URL, ERROR_MESSAGES */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+
+		const clone = event.request.clone();
+		return fetch( event.request )
+			.then( ( response ) => {
+				return response;
+			} )
+			.catch( () => {
+				const bodyPromise = clone.blob();
+				bodyPromise.then(
+					function( body ) {
+						const request = event.request;
+						const req = new Request( request.url, {
+							method: request.method,
+							headers: request.headers,
+							mode: 'same-origin',
+							credentials: request.credentials,
+							referrer: request.referrer,
+							redirect: 'manual',
+							body: body
+						} );
+
+						// Add request to queue.
+						queue.addRequest( req );
```

Comment:
Per GoogleChrome/workbox#1710 `addRequest` is going to be replaced in v4.

Reply:
We could update to v4 now if that is helpful.

Reply:
If we're going to update it anyway then probably would be good to do it now.